Anonymize/Anonymizer.php

```diff
@@ -22,38 +26,53 @@
 class Anonymizer
 {
     /**
-     * The annotation reader service
-     *
      * @var AnnotationReader
      */
     private $annotationReader;
 
+    /**
+     * @var PropertyAnonymizer
+     */
+    private $propertyAnonymizer;
+
     /**
      * Anonymizer constructor.
      *
-     * @param AnnotationReader $annotationReader The annotation reader service
+     * @param AnnotationReader   $annotationReader   The annotation reader that should be used
+     * @param PropertyAnonymizer $propertyAnonymizer The property anonymizer
      */
-    public function __construct(AnnotationReader $annotationReader)
-    {
-        $this->annotationReader = $annotationReader;
+    public function __construct(
+        AnnotationReader $annotationReader,
+        PropertyAnonymizer $propertyAnonymizer
+    ) {
+        $this->annotationReader     = $annotationReader;
+        $this->propertyAnonymizer   = $propertyAnonymizer;
     }
 
     /**
-     * Anonymize the object passed
+     * Anonymizes the given object which should contain the @see Anonymize annotations.
      *
-     * @param object $object The object that is being anonymized
+     * @param object $object The object to anonymize.
      *
-     * @todo implement function
+     * @throws InvalidArgumentException
+     * @throws ReflectionException
      *
      * @return void
      */
-    public function anonymize($object)
+    public function anonymize(/*object */$object)
```

Comment:
What is the reason behind `/*object */`?

Reply:
In PHP 7 that typehint is possible, but not yet in 5.6 so for now it's commented out.

Reply:
is it then maybe better to get rid of it and use that typehint when we use php7 as minimum?

Reply:
This has been discussed before and the outcome was this comment in the argument.

So PHP 5 can't handle object type hints and we want to be as strict as possible, the conclusion was to put a comment /* object */ with the argument and an is_object check in the function. When we are going to switch to PHP7 this will be replaced with the strict type object hint.